control-plane-operator/controllers/hostedcontrolplane/cco/reconcile.go

```diff
@@ -0,0 +1,454 @@
+			startScript,
+		}
+		c.Env = []corev1.EnvVar{
+			{
+				Name:  "RELEASE_VERSION",
+				Value: releaseVersion,
+			},
+			{
+				Name:  "KUBECONFIG",
+				Value: path.Join(volumeMounts.Path(c.Name, volumeClientToken().Name), kubeconfigFile),
+			},
+		}
+		proxy.SetEnvVars(&c.Env)
+		c.VolumeMounts = volumeMounts.ContainerMounts(c.Name)
+		c.TerminationMessagePolicy = corev1.TerminationMessageFallbackToLogsOnError
+		c.SecurityContext = &corev1.SecurityContext{
+			Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{corev1.Capability("ALL")}},
+			AllowPrivilegeEscalation: pointer.Bool(false),
+		}
+	}
+}
+
+func containerMetrics() *corev1.Container {
+	return &corev1.Container{
+		Name: "metrics",
+	}
+}
+
+func renderProxyStartScript(startScriptTemplate *template.Template) (string, error) {
+	out := &bytes.Buffer{}
+	params := struct {
+		TokenDir    string
+		TokenFile   string
+		MetricsPort int
+		CertFile    string
+		KeyFile     string
+	}{
+		TokenDir:    volumeMounts.Path(containerMetrics().Name, volumeClientToken().Name),
+		TokenFile:   kubeconfigFile,
+		MetricsPort: metricsPort,
+		CertFile:    path.Join(volumeMounts.Path(containerMetrics().Name, volumeServingCert().Name), "tls.crt"),
+		KeyFile:     path.Join(volumeMounts.Path(containerMetrics().Name, volumeServingCert().Name), "tls.key"),
+	}
+	if err := startScriptTemplate.Execute(out, params); err != nil {
+		return "", err
+	}
+	return out.String(), nil
+}
+
+func buildMetricsContainer(image string) func(*corev1.Container) {
+	return func(c *corev1.Container) {
+		c.Image = image
+		c.Command = []string{
+			"/bin/bash",
+		}
+		c.Args = []string{
+			"-ec",
+			proxyStartScript,
+		}
+		c.Ports = []corev1.ContainerPort{
+			{
+				ContainerPort: metricsPort,
+				Name:          "metrics",
+				Protocol:      corev1.ProtocolTCP,
+			},
+		}
+		c.VolumeMounts = volumeMounts.ContainerMounts(c.Name)
+		c.TerminationMessagePolicy = corev1.TerminationMessageReadFile
+		c.TerminationMessagePath = "/dev/termination-log"
+		c.SecurityContext = &corev1.SecurityContext{
+			Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{corev1.Capability("ALL")}},
+			AllowPrivilegeEscalation: pointer.Bool(false),
+		}
+	}
+}
+
+func containerClientTokenMinter() *corev1.Container {
+	return &corev1.Container{
+		Name: "client-token-minter",
+	}
+}
+
+func buildClientTokenMinter(image, issuerURL string) func(*corev1.Container) {
+	return func(c *corev1.Container) {
+		c.Image = image
+		c.Command = []string{
+			"/usr/bin/control-plane-operator",
+			"token-minter",
+		}
+		c.Args = []string{
+			"--service-account-namespace",
+			workerNamespace,
+			"--service-account-name",
+			workerServiceAccount,
+			"--token-file",
+			path.Join(volumeMounts.Path(c.Name, volumeClientToken().Name), kubeconfigFile),
+			"--token-audience",
+			issuerURL,
+			"--kubeconfig",
+			path.Join(volumeMounts.Path(c.Name, volumeAdminKubeconfig().Name), kas.KubeconfigKey),
```

Comment:
why admin?

Reply:
If you have a different token to use, please let me know. I am following the example of the registry operator here.